R/regularized_regression.R

```diff
@@ -409,3 +409,74 @@ mrash_weights <- function(X, y, init_prior_sd = TRUE, ...) {
   fit.mr.ash <- do.call("mr.ash", c(list(X = X, y = y, sa2 = if (init_prior_sd) init_prior_sd(X, y)^2 else NULL), args_list))
   predict(fit.mr.ash, type = "coefficients")[-1]
 }
+#' Extract Coefficients From Bayesian Linear Regression
+#'
+#' This function performs Bayesian linear regression using the `gbayes` function from 
+#' the `qgg` package. It then returns the estimated slopes.
+#'
+#' @param y A numeric vector of phenotypes.
+#' @param X A numeric matrix of genotypes.
+#' @param method A character string declaring the method/prior to be used. Options are 
+#' bayesN, bayesL, bayesA, bayesC, or bayesR.
+#' @param Z An optional numeric matrix of covariates.
+#' @return A vector containing the weights to be applied to each genotype in 
+#'   predicting the phenotype.
+#' @details This function fits a Bayesian linear regression model 
+#' @examples
+#' X <- matrix(rnorm(100000), nrow=1000)
+#' Z <- matrix(round(runif(3000, 0, 0.8), 0), nrow=1000)
+#' set1 <- sample(1:ncol(W), 5)
+#' set2 <- sample(1:ncol(W), 5)
+#' sets <- list(set1, set2)
+#' g <- rowSums(W[, c(set1, set2)])
+#' e <- rnorm(nrow(W), mean=0, sd=1)
+#' y <- g + e
+#' bayes_l_weights(y=y, X=X, Z=Z)
+#' bayes_r_weights(y=y, X=X, Z=Z)
+#' @importFrom qgg gbayes
+#' @export
+bayes_alphabet_weights <- function(y, X, method, Z=NULL){
```

Comment:
Our interface https://github.com/cumc/pecotmr/blob/main/R/regularized_regression.R#L384 the input should be X, y not y, X

Reply:
All functions have been updated to X, y format.